src/snapshotter/snapshotter.ts

```diff
@@ -62,68 +62,94 @@ export class Snapshotter {
       return
     }
 
-    this.logger.info(`Running cleanup before snapshot`)
+    this.logger.info(`Running presave before snapshot`)
     const pwd = process.cwd()
     this.logger.debug(`Current working directory: ${pwd}`)
 
-    const cleanupScript = `
+    const presaveScript = `
 #!/bin/bash
 
 set -e
 
+echo "Saving environment variables to /etc/environment"
+
+printenv | grep -v '^GITHUB_' | grep -v '^WARP_' | grep -v '^ACTIONS_' | grep -v '^INPUT_' | grep -v '^WARPBUILD_' | grep -v '^RUNNER_' | grep -v '^_' | grep -v '^USER$' | grep -v '^INVOCATION_ID$' | grep -v '^SYSTEMD_EXEC_PID$' | grep -v '^SUDO_USER$' | grep -v '^SUDO_GID$' | grep -v '^SUDO_UID$' | grep -v '^SHELL$' | grep -v '^SHLVL$' | grep -v '^XDG_' | grep -v '^JOURNAL_STREAM$' | grep -v '^HOME$' | sort | while IFS= read -r line; do
+  key=$(echo "$line" | cut -d= -f1)
+  value=$(echo "$line" | cut -d= -f2-)
+  if grep -q "^\${key}=" /etc/environment; then
+    sudo sed -i "s|^\${key}=.*|\${key}=\${value}|" /etc/environment
+  else
+    printf '%s=%q\n' "$key" "$value" | sudo tee -a /etc/environment > /dev/null
+  fi
+done
+
+echo "Logging environment variables in /etc/environment"
+echo "--------------------------------"
+cat /etc/environment
+echo "--------------------------------"
+
+# Remove PATH from /runner/.env
+sed -i '/^PATH=/d' /runner/.env
```

Comment:
IIRC, this PATH is used by GitHub runner when spinning up the job. Will this be correctly populated next time the new VM spins up?

Reply:
Yes. PATH is being set properly